functions.php

```diff
@@ -198,93 +198,32 @@ function hide_admin_menu() {
-		foreach ($items as $key => $item) {
-			$html .= format_item($item);
-		}
-		return $html;
-
-	}
-
-	// This is a copy (with some front page changes) of the views/items_blades/title_blades, if you should ever decide to rewrite & combine
-
-	function format_item($item)
-	{
-
-		$html = '<li class="catalog-result">';
-
-		$image = (empty($item['coverart_thumbnail']))? LIBRIVOX_CATALOG_URL .'images/book-cover-default-65x65.gif': $item['coverart_thumbnail'] ;
-		//$image = ($item['status'] == 'complete') ? $image  : LIBRIVOX_CATALOG_URL .'images/book-cover-in-progress-65x65.gif';			
-
-		$html .= '<a href="' . $item['url_librivox'] . '" class="book-cover"><img src="'. $image .'" alt="book-cover-65x65" width="65" height="65" /></a>';
-
-		$authors = author_list($item['authors']);
-
-		$html .= '<div class="result-data">
-					<h3><a href="' . $item['url_librivox'] .' ">'.create_full_title($item).'</a></h3>
-					<p class="book-author"> ' . $authors . '</p>
-					<p class="book-meta"> ' . 'Complete '. ' | ' . ucwords($item['project_type']) . ' | ' . $item['language'] . '</p>
-				</div>';	
-
-		$html .= '</li>';
-
-		return $html;
-
-	}
-
-	function create_full_title($project)
-	{
-		if (is_array($project)) $project = (object) $project;
-
-		$project->full_title = (!empty($project->title_prefix))? $project->title_prefix.' ': '';    
-	    return $project->full_title .= $project->title;
-	}
-
-	function author_list($author_list='')
-	{
-
-		if (empty($author_list)) return '';
-
-		foreach ($author_list as $key => $author) {
-			$authors[] = '<a href="'.LIBRIVOX_CATALOG_URL.'author/'.$author['id'].'">'.$author['author'] . ' <span class="dod-dob">' . build_author_years((object) $author) .'</span></a>';  
-		}
-		return implode(', ' , $authors);
-	}
-
-	function build_author_years($author)
-	{
-		$author->dob = (empty($author->dob)) ? ' ': $author->dob;
-		$author->dod = (empty($author->dod)) ? ' ': $author->dod;
-
-		return sprintf("(%s)", implode(' - ', array($author->dob, $author->dod)));
-
-	}
-
-	// not in use, but we may want it back - our WP git repo is a little "unreliable"
-	function create_days_ago($project)
-	{
-		$project = (object) $project;
-
-		$date = new DateTime($project->date_catalog);
-		$date->format('Y-m-d');
-
-		$now = new DateTime();
-
-		$interval = $date->diff($now);
-		//var_dump($interval);
-
-		if ($interval->days < 1) $days_ago = 'Released today';
-		if ($interval->days == 1) $days_ago = 'Released yesterday';
-		if ($interval->days > 1) $days_ago = 'Released ' . $interval->days . ' days ago';
-
-		return $days_ago; 
-
+// Librivox Catalog Feed items
+function format_results($items)
+{
+	$html = '';
+	foreach ($items as $key => $item) {
+		$html .= format_item($item);
 	}
+	return $html;
+}
 
-
-?>
\ No newline at end of file
+// Lets try to re-use as much formatting code from catalog as possible such as
+// create_full_title and format_authors.
+require_once('../catalog/application/helpers/previewer_helper.php');
```

Comment:
I instinctively don't like this, though I'd have trouble putting into words exactly why. Something about coupling and dependencies, I suppose.

That being said, let's be honest with ourselves - the WordPress theme and catalog are two halves of the same web application. The separation is due to the logistics rather smart design. So I think I'm OK with this, especially since I don't have any better ideas that don't involve tonnes of pointless work.

Reply:
In that case I will remove the draft status. I have also prepared a PR for the catalog code to add comments warning of the dependency.